### ISSUE TITLE
ocsp-update not working if issuers-chain-path is used for intermediates

### ISSUE THREAD

**Issue by amsnek** [haproxy/haproxy#2635]
### Detailed Description of the Problem

When using the feature [ocsp-update on] it does not work if the intermediate certificates are used seperately with an  intermediate certificate store (issuers-chain-path). Attemping to update OCSP will result in "OCSP response check failure". 
When the intermediate for the certificate is added to certificate.pem -> it works as intended. 

when running in debug it shows that the ocsp response is retrieved successfully, but fails nonetheless, probably due to verification problems
```
00000000:<OCSP-UPDATE>.srvrep[ffffffff:0016]: HTTP/1.1 200 OK
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: date: Wed, 10 Jul 2024 10:56:07 GMT
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: content-type: application/ocsp-response
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: content-length: 282
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: last-modified: Wed, 10 Jul 2024 05:34:03 GMT
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: expires: Wed, 17 Jul 2024 05:34:02 GMT
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: etag: "8671a2279993a1e298afeb44bda705a27fcfe5fd"
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: cache-control: max-age=603286,s-maxage=1800,public,no-transform,must-revalidate
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: x-ccacdn-proxy-id: mcdpinlb4
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: x-frame-options: SAMEORIGIN
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: cf-cache-status: HIT
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: age: 1137
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: accept-ranges: bytes
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: server: cloudflare
00000000:<OCSP-UPDATE>.srvhdr[ffffffff:0016]: cf-ray: 8a10067e792dc31c-VIE
```

### Expected Behavior

[ocsp-update on] should also work if intermediates are placed in a seperate folder instead the certificate.pem itself

### Steps to Reproduce the Behavior

1) configure "use issuers-chain-path /data/haproxy/ssl/intermediates" in the global section
2) use for example ancrt-list with /data/haproxy/ssl/certificate.pem [alpn h2 ocsp-update on] 
3) reload haproxy 

### Do you have any idea what may have caused this?

ocsp response verification is not working due to intermediates being seperate/assembled by haproxy. 

### What is your configuration?

```haproxy
global
  issuers-chain-path /data/haproxy/ssl/intermediates/

frontend ocsp-test
  bind 127.0.0.1:443  ssl crt-list /data/haproxy/ssl/crt-list_OCSP-test.txt

/data/haproxy/ssl/crt-list_OCSP-test.txt:
/data/haproxy/ssl/certficate.pem [alpn h2 ocsp-update on]
```

### Output of `haproxy -vv`

```plain
HAProxy version 3.0.2-a45a8e6 2024/06/14 - https://haproxy.org/
Status: long-term supported branch - will stop receiving fixes around Q2 2029.
Known bugs: http://www.haproxy.org/bugs/bugs-3.0.2.html
Running on: Linux 4.18.0-553.5.1.el8_10.x86_64 #1 SMP Tue May 21 03:13:04 EDT 2024 x86_64
Build options :
  TARGET  = linux-glibc
  CC      = cc
  CFLAGS  = -O2 -g -fwrapv -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Werror=format-security
  OPTIONS = USE_OPENSSL=1 USE_LUA=1 USE_SLZ=1 USE_TFO=1 USE_NS=1 USE_SYSTEMD=1 USE_QUIC=1 USE_PROMEX=1 USE_STATIC_PCRE2=1 USE_PCRE2_JIT=1 USE_QUIC_OPENSSL_COMPAT=1
  DEBUG   =

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT -PCRE +PCRE2 +PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL +PROMEX -PTHREAD_EMULATION +QUIC +QUIC_OPENSSL_COMPAT +RT +SHM_OPEN +SLZ +SSL -STATIC_PCRE +STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL -ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=2).
Built with OpenSSL version : OpenSSL 3.3.1 4 Jun 2024
Running on OpenSSL version : OpenSSL 3.3.1 4 Jun 2024
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
OpenSSL providers loaded : default
Built with Lua version : Lua 5.4.7
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with libslz for stateless compression.
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.44 2024-06-07
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with gcc compiler version 8.5.0 20210514 (Red Hat 8.5.0-22)

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
       quic : mode=HTTP  side=FE     mux=QUIC  flags=HTX|NO_UPG|FRAMED
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [BWLIM] bwlim-in
        [BWLIM] bwlim-out
        [CACHE] cache
        [COMP] compression
        [FCGI] fcgi-app
        [SPOE] spoe
        [TRACE] trace
```

**Comment by einval22** [comment]
Hi amsnek ! 

Thanks for reporting this. It seems the case, which we've have not covered yet for ocsp-updates.

I'm reproducing it.
Would you please, dump here, just to be sure,  the output of  

`echo "show ssl ocsp-updates" | socat <path_to stats_sock>  stdio`

and also 

`echo "show ssl ocsp-response" | socat <path_to_stats_sock> stdio`

If you run haproxy in master-worker mode, just to prefix this "show..." commands with @1 to send them to worker process

Kind regards,

**Comment by amsnek** [comment]
Hello vkssv!

Thank you for looking into this. Attached the output (slightly obfuscated)

echo "show ssl ocsp-updates" | socat /run/haproxy.sock  stdio

```
OCSP Certid | Path | Next Update | Last Update | Successes | Failures | Last Update Status | Last Update Status (str)
0906052b0304a30e03021a0500041478cd37357eae292bd9aabab63f4523aa11a7953f80c2cd4d39120414edb4a0336a196c8bf4e8200b0891b6bdfa413021100fc1feb776f5a24ed54915c4 | /data/haproxy/ssl/certificate.pem | 11/Jul/2024:13:00:53 +0200 | - | 0 | 1 | 4 | OCSP response check failure
```

echo "show ssl ocsp-response" | socat /run/haproxy.sock stdio

```
# Certificate IDs
Certificate ID key : 0906052b0304a30e03021a0500041478cd37357eae292bd9aabab63f4523aa11a7953f80c2cd4d39120414edb4a0336a196c8bf4e8200b0891b6bdfa413021100fc1feb776f5a24ed54915c4
Certificate path : /data/haproxy/ssl/certificate.pem
 Certificate ID:
   Issuer Name Hash: 96C88200AA11A7953F80C2CD4D391278CD23BF4E
   Issuer Key Hash: 336A11B6BDFA4192BD9AABAB63F453EDB4A0B089
   Serial Number: B7764ED54917EAE25C4FC1FE3735F5A2
```

**Comment by einval22** [comment]
Yes thanks !
I've reproduced on my side as well ! 
I'm looking how to fix, keep you updated on this.

Thanks a lot for reporting the problem !

**Comment by wlallemand** [comment]
Hello,

Thanks to @vkssv tests and analysis I could take a look at the problem. The 'issuers-chain-path' keyword was not designed to be used with the OCSP loading . 

I could make the change so this would be compatible, but it changes the ckch_data structure so I'm not very confortable with backporting it to previous versions.  This will be compatible in future 3.1 version though.

Regards,

**Comment by wlallemand** [comment]
I'm closing this, this won't be backported, feature will be available in the future 3.1 release.
